src/axom/inlet/Table.hpp

```diff
@@ -277,52 +291,82 @@ class Table : public SchemaCreator, public std::enable_shared_from_this<Table>
    *****************************************************************************
-   * \brief Get a int array represented as an unordered map from the input deck
+   * \brief Add an array of Fields to the input deck schema.
    *
-   * \param [out] map Unordered map to be populated with array contents
+   * \param [in] name Name of the array
+   * \param [in] description Description of the Field
    *
-   * \return Whether or not the array was found
+   * \return Shared pointer to the created Field
    *****************************************************************************
    */
-  bool getArray(std::unordered_map<int, int>& map);
+  std::shared_ptr<Table> addGenericArray(const std::string& name,
+                                         const std::string& description = "");
 
   /*!
    *****************************************************************************
-   * \brief Get a double array represented as an unordered map from the input deck
+   * \brief Get an array represented as an unordered map from the input deck
+   * of primitive type
    *
    * \param [out] map Unordered map to be populated with array contents
    *
    * \return Whether or not the array was found
    *****************************************************************************
    */
-  bool getArray(std::unordered_map<int, double>& map);
+  template <typename T>
+  typename std::enable_if<detail::is_inlet_primitive<T>::value, bool>::type
+  getArray(std::unordered_map<int, T>& map)
+  {
+    map.clear();
+    if(!axom::utilities::string::endsWith(m_name, "_inlet_array"))
+    {
+      return false;
+    }
+    for(auto& item : m_fieldChildren)
+    {
+      auto pos = item.first.find_last_of("/");
+      int index = std::stoi(item.first.substr(pos + 1));
+      map[index] = item.second->get<T>();
+    }
+    return true;
+  }
 
   /*!
    *****************************************************************************
-   * \brief Get a string array represented as an unordered map from the input deck
+   * \brief Get an array represented as an unordered map from the input deck
+   * of user-defined type
    *
    * \param [out] map Unordered map to be populated with array contents
    *
    * \return Whether or not the array was found
    *****************************************************************************
    */
-  bool getArray(std::unordered_map<int, std::string>& map);
+  template <typename T>
+  typename std::enable_if<!detail::is_inlet_primitive<T>::value, bool>::type
+  getArray(std::unordered_map<int, T>& map)
+  {
+    if(m_sidreGroup->hasView("_inlet_array_indices"))
+    {
+      auto view = m_sidreGroup->getView("_inlet_array_indices");
+      int* array = view->getArray();
+      for(int i = 0; i < view->getNumElements(); i++)
+      {
+        auto index_label = std::to_string(array[i]);
+        map[array[i]] = getTable(index_label)->get<T>();
+      }
+    }
+    else
+    {
+      SLIC_WARNING("Table does not contain an array");
```

Comment:
prefix this message with "[Inlet]" since some of these messages are pretty generic sounding without it.